src/script/campaign/mod/vco-listeners.lua

```diff
@@ -2,6 +2,29 @@ local vco = core:get_static_object("vco");
 
 -- COMMON --
 
+-- POOLED RESOURCES --
+
+local function update_faction_pooled_resource_earnings(faction, pooled_resource_key, amount)
+  local amount_of_resource_earned = cm:get_saved_value("vco_pooled_resource_earned_" .. faction .. "_" .. pooled_resource_key) or 0;
+
+  if amount > 0 then
+    cm:set_saved_value(
+      "vco_pooled_resource_earned_" .. faction .. "_" .. pooled_resource_key,
+      amount_of_resource_earned + amount
+    );
+    return true
```

Comment:
We do not need to return anything here. Just the `cm:set_saved_value` part is grand.